src/devices/Rtc/RtcBase.cs

```diff
@@ -10,15 +10,57 @@ namespace Iot.Device.Rtc
     /// </summary>
     public abstract class RtcBase : IDisposable
     {
+        private TimeZoneInfo _timeZone;
+
         /// <summary>
-        /// The Device's <see cref="System.DateTime"/>
+        /// The Device's raw <see cref="System.DateTime"/>.
+        /// The caller must be aware of the device's time zone.
+        /// The behavior of the <see cref="System.DateTime.Kind"/> property is implementation-dependent (typically it is ignored)
         /// </summary>
-        public virtual DateTime DateTime
+        public virtual DateTime RtcDateTime
         {
             get => ReadTime();
             set => SetTime(value);
         }
 
+        /// <summary>
+        /// Set or retrieves the current date/time. This property returns a <see cref="DateTimeOffset"/> and
+        /// is therefore correct regardless of the current time zone (when <see cref="TimeZone"/> is set correctly).
+        /// </summary>
+        public DateTimeOffset DateTime
+        {
+            get
+            {
+                var now = RtcDateTime;
+                return new DateTimeOffset(now.Ticks, TimeZone.GetUtcOffset(now));
+            }
+            set
+            {
+                var clockNow = new DateTime((value.UtcDateTime + TimeZone.GetUtcOffset(value)).Ticks, DateTimeKind.Local);
+                RtcDateTime = clockNow;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the time zone this clock operates in.
+        /// Defaults to the local time zone.
+        /// Changing this property will not change the time on the real time clock,
+        /// but instead affect the return value of <see cref="DateTime"/>
+        /// </summary>
+        public virtual TimeZoneInfo TimeZone
```

Comment:
This should be clear in the property name if this is RtcTimeZone or LocalTimeZone